drv/ignition-server/src/main.rs

```diff
@@ -210,6 +205,30 @@ impl idl::InOrderIgnitionImpl for ServerImpl {
             .map_err(IgnitionError::from)
             .map_err(RequestError::from)
     }
+
+    fn state_dump(
```

Comment:
You _could_ implement this as a HIF loop on the Humility side, e.g. something like [this](https://github.com/oxidecomputer/humility/blob/master/cmd/monorail/src/lib.rs#L922-L948).

Reply:
That was actually my first idea, but after some back and forth with @bcantrill about the intent/limitations of HIF I decided to add this function which can also be used by the control plane/technician port functions down the road.